frontend/image.py

```diff
@@ -40,4 +51,7 @@ def set_alpha(self, alpha):
         self.image.set_alpha(alpha)
 
     def set_image(self, image):
+        """
+        """
+        self.source_image = image
```

Comment:
note to self - this is necessary to avoid repeated smooth scaling operations on the same image eventually making it transparent. I'd add a comment in the code too.